configs/components/facter.rb

```diff
@@ -134,6 +144,13 @@
 
     # FACT-1156: If we build with -O3, solaris segfaults due to something in std::vector
     special_flags = "-DCMAKE_CXX_FLAGS_RELEASE='-O2 -DNDEBUG'"
+  elsif platform.is_windows?
+    make = "#{settings[:gcc_bindir]}/mingw32-make"
+    pkg.environment "PATH" => "$$(cygpath -u #{settings[:gcc_bindir]}):$$(cygpath -u #{settings[:bindir]}):/cygdrive/c/Windows/system32:/cygdrive/c/Windows:/cygdrive/c/Windows/System32/Wbem:/cygdrive/c/Windows/System32/WindowsPowerShell/v1.0:/cygdrive/c/pstools"
```

Comment:
Yeah, that would be `$$PATH`. We don't get that luxury here though. Since we're building for MinGW Makefiles in a cygwin environment, we can't have cygwin sh.exe on the path. Since it's there by default in the cygwin environment, we need to overwrite the entire path :( I guess I could look at just removing the cygwin specific paths, but that seemed too complicated to do